test/unit/specs/components/VHeader/v-filter-button.spec.js

```diff
@@ -0,0 +1,140 @@
+    localVue.use(VueI18n)
+
+    storeMock = new Vuex.Store({
+      modules: {
+        search: {
+          namespaced: true,
+          getters: {
+            isAnyFilterApplied: () => appliedFilters.length > 0,
+            appliedFilterTags: () => appliedFilters,
+          },
+        },
+      },
+    })
+    options = {
+      localVue,
+      propsData: props,
+      mocks: {
+        $nuxt: {
+          context: {
+            store: storeMock,
+            i18n,
+          },
+        },
+      },
+    }
+  })
+
+  describe('Above the medium breakpoint', () => {
+    it('always shows the label and icon', () => {
+      isScreen.mockImplementation(() => ({ value: true }))
+      const { container } = render(VFilterButton, options)
+
+      const button = screen.getByText('Filters')
+      const icon = container.querySelector('svg')
+
+      expect(button).toBeVisible()
+      expect(icon).toBeVisible()
+    })
+    it('shows the count and text when filters are applied', () => {
+      isScreen.mockImplementation(() => ({ value: true }))
+      // +2 to guarantee it's plural
+      const filterCount = Math.floor(Math.random() * 10) + 2
+      appliedFilters = Array(filterCount).fill('')
+      render(VFilterButton, options)
+
+      const button = screen.getByText(`${filterCount} Filters`)
+
+      expect(button).toBeVisible()
+    })
+    it('does not show the icon when filters are applied', () => {
+      isScreen.mockImplementation(() => ({ value: true }))
+      appliedFilters = ['mockfilter1', 'mockfilter2']
+
+      const { container } = render(VFilterButton, options)
+      const icon = container.querySelector('svg')
+
+      expect(icon).not.toBeInTheDocument()
+    })
+  })
+
+  describe('below the medium breakpoint', () => {
+    it('only shows the filter icon by default', () => {
+      isScreen.mockImplementation(() => ({ value: false }))
+      appliedFilters = []
+      const { container } = render(VFilterButton, options)
+
+      const icon = container.querySelector('svg')
+      const label = screen.queryByTestId('filterbutton-label')
+
+      expect(icon).toBeVisible()
+      expect(label).not.toBeInTheDocument()
+    })
+    it('only shows the count and label when filters are applied', () => {
+      isScreen.mockImplementation(() => ({ value: false }))
+      // +2 to guarantee it's plural
+      const filterCount = Math.floor(Math.random() * 10) + 2
+      appliedFilters = Array(filterCount).fill('')
+      const { container } = render(VFilterButton, options)
+
+      const icon = container.querySelector('svg')
+      const button = screen.getByText(`${filterCount} Filters`)
+
+      expect(icon).not.toBeInTheDocument()
+      expect(button).toBeVisible()
+    })
+    it('only shows the count when filters are applied and the user scrolls', () => {
+      isScreen.mockImplementation(() => ({ value: false }))
+      // +2 to guarantee it's plural
+      const filterCount = Math.floor(Math.random() * 10) + 2
```

Comment:
Curious what you think of this, but in my experience introducing randomness into tests that only run once per suite (as opposed to something like property based testing which would more rigorously test the input/output based on random and vast examples) run can introduce difficult to debug flakiness.

By virtue of that we tend to expect the randomness to cover all the cases we'd have to manually test otherwise, but using `it.each` we can easily repeat tests for different cases and test specific edge cases.

For example, I'd probably recommend testing `[0, 1, 2, 3, 4]` to cover pluralization rules in most languages (though admitedly not all). Right now we're not testing 0 and 1 in the scrolled state at all, which are distinct branches.

Reply:
That makes sense. I mainly wanted to protect against some specific test where only the hardcoded number of filters worked, but the `it.each` approach sounds lovely. I'm going to add those tests for pluralization before merging 👍